modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDataTypesTest.java

```diff
@@ -86,6 +87,23 @@ public void testUnicodeStrings() {
         }
     }
 
+    /** Test correctness of char types limitation. */
+    @Test
+    public void charLimitation() {
+        sql("create table limitedChar (pk int primary key, f1 VARCHAR(5), f2 CHAR(4))");
+        assertThrows(CalciteContextException.class, () -> sql("insert into limitedChar(pk, f2) values (1, 'aaaadddd')"));
+        assertThrows(CalciteContextException.class, () -> sql("insert into limitedChar values (2, 'aaaaabbbbb', 'aaaadddd')"));
```

Comment:
```suggestion
        assertThrows(CalciteContextException.class, () -> sql("insert into limitedChar(pk, f1) values (2, 'aaaaabbbbb')"));
```
Let's check CHAR and VARCHAR columns independently.

Reply:
It would be even better to have a method not only excepts some generic CalciteContextException but also checks for error message.